File: Task_Align_Frontend/src/services/authService.js
import apiClient from "./apiClient.js";

/**
 * Authentication Service
 * Interfaces directly with Task_Align_Backend /api/auth/* REST endpoints.
 */
export const authService = {
  /**
   * User Email/Password Login
   * @param {Object} payload { email, password, rememberme }
   * @returns {Promise<Object>} LoginResponse data
   */
  async login(payload) {
    const res = await apiClient.post("/api/auth/login", payload);
    // Backend returns ApiResponse<LoginResponse> { success, message, data: { userId, fullName, email, mobileNumber, token, tokenType } }
    return res.data?.data || res.data;
  },

  /**
   * Register User
   * @param {Object} payload { fullName, email, mobileNumber, gender, state, country, password, confirmPassword }
   * @returns {Promise<Object>} RegisterResponse data
   */
  async register(payload) {
    const res = await apiClient.post("/api/auth/register", payload);
    return res.data?.data || res.data;
  },

  /**
   * Send Mobile OTP
   * @param {Object} payload { mobileNumber }
   * @returns {Promise<string>} Success message
   */
  async sendMobileOtp(payload) {
    const res = await apiClient.post("/api/auth/send-mobile-otp", payload);
    return res.data?.data || res.data;
  },

  /**
   * Mobile Login with OTP
   * @param {Object} payload { mobileNumber, otp }
   * @returns {Promise<Object>} LoginResponse data
   */
  async mobileLogin(payload) {
    const res = await apiClient.post("/api/auth/mobile-login", payload);
    return res.data?.data || res.data;
  },

  /**
   * Forgot Password - Send OTP to Email
   * @param {Object} payload { email }
   * @returns {Promise<string>} Success message
   */
  async forgotPassword(payload) {
    const res = await apiClient.post("/api/auth/forgot-password", payload);
    return res.data?.data || res.data;
  },

  /**
   * Verify OTP for Password Reset
   * @param {Object} payload { email, otp }
   * @returns {Promise<string>} Success message
   */
  async verifyOtp(payload) {
    const res = await apiClient.post("/api/auth/verify-otp", payload);
    return res.data?.data || res.data;
  },

  /**
   * Reset Password
   * @param {Object} payload { email, otp, newPassword, confirmPassword }
   * @returns {Promise<string>} Success message
   */
  async resetPassword(payload) {
    const res = await apiClient.post("/api/auth/reset-password", payload);
    return res.data?.data || res.data;
  },

  /**
   * Edit User Profile (Protected)
   * @param {Object} payload { fullName, mobileNumber, state, country }
   * @returns {Promise<Object>} ProfileResponse data
   */
  async editProfile(payload, profilePicture = null) {
  const formData = new FormData();

  formData.append("fullName", payload.fullName);
  formData.append("mobileNumber", payload.mobileNumber);
  formData.append("gender", payload.gender);
  formData.append("state", payload.state);
  formData.append("country", payload.country);

  if (profilePicture) {
    formData.append("profilePicture", profilePicture);
  }

  const res = await apiClient.put(
    "/api/auth/edit-profile",
    formData
  );

  return res.data?.data || res.data;
},

  /**
   * Change Password (Protected)
   * @param {Object} payload { currentPassword, newPassword, confirmPassword }
   * @returns {Promise<string>} Success message
   */
  async changePassword(payload) {
    const res = await apiClient.put("/api/auth/change-password", payload);
    return res.data?.data || res.data;
  },
  /**
 * Get Current User Profile (Protected)
 * @returns {Promise<Object>} ProfileResponse data
 */
async getProfile() {
  const res = await apiClient.get("/api/auth/profile");
  return res.data?.data || res.data;
},
};

File: Task_Align_Backend/src/main/java/com/task/www/service/AuthService.java
package com.task.www.service;

import com.task.www.dto.ChangePasswordRequest;
import com.task.www.dto.EditProfileRequest;
import com.task.www.dto.ForgotPasswordRequest;
import com.task.www.dto.LoginRequest;
import com.task.www.dto.MobileLoginRequest;
import com.task.www.dto.RegisterRequest;
import com.task.www.dto.ResetPasswordRequest;
import com.task.www.dto.VerifyOtpRequest;
import com.task.www.dto.ApiResponse;
import com.task.www.dto.LoginResponse;
import com.task.www.dto.ProfileResponse;
import com.task.www.dto.RegisterResponse;

public interface AuthService {

    ApiResponse<RegisterResponse> register(RegisterRequest request);

    ApiResponse<LoginResponse> login(LoginRequest request);
    
    ApiResponse<String> sendMobileOtp(String mobileNumber);

    ApiResponse<LoginResponse> mobileLogin(MobileLoginRequest request);
    
    ApiResponse<String> sendForgotPasswordOtp(String email);
    
    ApiResponse<String> verifyOtp(VerifyOtpRequest request);
    
    ApiResponse<String> resetPassword(ResetPasswordRequest request);

    ApiResponse<ProfileResponse> editProfile(EditProfileRequest request,
            String email);
    
    ApiResponse<String> changePassword(ChangePasswordRequest request,
            String email);
    
    ApiResponse<ProfileResponse> getProfile(String email);

    
}
